packages/ui-extensions-sdk/src/utils/security.ts

```diff
@@ -42,44 +42,36 @@ export const parseResource = (
     };
 };
 
+const collectInitialResources = () =>
+    performance
+        .getEntriesByType('resource')
+        .filter(isResourceTimingEntry)
+        .map(parseResource);
+
 export const startResourceMonitoring = (
     reportUsage: (batch: LoadedResourceMetaDataBatch) => void
 ): (() => void) => {
     if (!isPerformanceObjectSupported()) {
         return () => {};
     }
 
+    // first, report all previously loaded resources
+    reportUsage({ resources: collectInitialResources() });
+
+    // now, set up observer for all new resources
     const observerCallback = (entries: PerformanceObserverEntryList) => {
         const resources = entries
             .getEntries()
             .filter(isResourceTimingEntry)
             .map(parseResource);
-        return reportUsage({ resources });
+
+        reportUsage({ resources });
     };
 
     const observer = new PerformanceObserver(observerCallback);
     observer.observe({ type: 'resource', buffered: true });
 
-    let bufferFullHandler = () => {};
-
-    if ('addEventListener' in performance) {
-        bufferFullHandler = () => {
-            performance.clearResourceTimings();
-        };
```

Comment:
@nmuldavin It's not clear to me how the `PerformanceObserver` will behave in case resource timing buffer is full. Reading at https://chromestatus.com/feature/5320666234486784 my understanding is that we could miss some entries if the buffer is full. If this is the case I wonder if we should keep the call to `clearResourceTimings` to handle that case ?

Reply:
Resolving my comment, from the [spec](https://w3c.github.io/performance-timeline) it seems Performance observer buffer is emptied after the observers have been notified, so no need to clear it.

Reply:
yeah i did some reading on this, i think we are ok. 

There is one possible edge-case, which is that the buffer fills before the observer is instantiated such that the initial call to `performance.getEntriesByType()` isn't complete, but this is very unlikely given that the SDK must be instantiated early